haystack/nodes/reader/farm.py

```diff
@@ -269,26 +239,213 @@ def train(
+                        "O0" (Normal FP32 training)
+                        "O1" (Mixed Precision => Recommended)
+                        "O2" (Almost FP16)
+                        "O3" (Pure FP16).
+                        See details on: https://nvidia.github.io/apex/amp.html
+        :param checkpoint_root_dir: the Path of directory where all train checkpoints are saved. For each individual
+               checkpoint, a subdirectory with the name epoch_{epoch_num}_step_{step_num} is created.
+        :param checkpoint_every: save a train checkpoint after this many steps of training.
+        :param checkpoints_to_keep: maximum number of train checkpoints to save.
+        :param caching whether or not to use caching for preprocessed dataset
+        :param cache_path: Path to cache the preprocessed dataset
+        :return: None
+        """
+        return self._training_procedure(data_dir=data_dir, train_filename=train_filename,
+        dev_filename=dev_filename, test_filename=test_filename,
+        use_gpu=use_gpu, batch_size=batch_size,
+        n_epochs=n_epochs, learning_rate=learning_rate,
+        max_seq_len=max_seq_len, warmup_proportion=warmup_proportion,
+        dev_split=dev_split, evaluate_every=evaluate_every,
+        save_dir=save_dir, num_processes=num_processes,
+        use_amp=use_amp, checkpoint_root_dir=checkpoint_root_dir,
+        checkpoint_every=checkpoint_every, checkpoints_to_keep=checkpoints_to_keep,
+        caching=caching, cache_path=cache_path)
+    
+    def distil_from(
+        self,
+        teacher_model: "FARMReader",
+        data_dir: str,
+        train_filename: str,
+        dev_filename: Optional[str] = None,
+        test_filename: Optional[str] = None,
+        use_gpu: Optional[bool] = None,
+        student_batch_size: int = 10,
+        teacher_batch_size: Optional[int] = None,
+        n_epochs: int = 2,
+        learning_rate: float = 1e-5,
+        max_seq_len: Optional[int] = None,
+        warmup_proportion: float = 0.2,
+        dev_split: float = 0,
+        evaluate_every: int = 300,
+        save_dir: Optional[str] = None,
+        num_processes: Optional[int] = None,
+        use_amp: str = None,
+        checkpoint_root_dir: Path = Path("model_checkpoints"),
+        checkpoint_every: Optional[int] = None,
+        checkpoints_to_keep: int = 3,
+        caching: bool = False,
+        cache_path: Path = Path("cache/data_silo"),
+        distillation_loss_weight: float = 0.5,
+        distillation_loss: Union[str, Callable[[torch.Tensor, torch.Tensor], torch.Tensor]] = "kl_div",
+        temperature: float = 1.0
+    ):
+        """
+        Fine-tune a model on a QA dataset using distillation. You need to provide a teacher model that is already finetuned on the dataset
+        and a student model that will be trained using the teacher's logits. The idea of this is to increase the accuracy of a lightweight student model
+        using a more complex teacher.
```

Comment:
Can you give a short code example here including a reasonable combination of models (to show usage and clarify that the student can be another pretrained model)?

Reply:
Okay